src/test/ui/consts/assoc_const_generic_impl.rs

```diff
@@ -0,0 +1,19 @@
+#![allow(const_err)]
```

Comment:
Is there a `const_err` lint reported now or just the hard error? If there is a lint, can we change this to warn so we can verify the warning is emitted?

Reply:
I am also worried about duplicate diagnostics here.